resolwe_bio/tools/star_sj_to_bed12.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/python3
+"""Recalculate from STAR SJ.out.tab file to BED12 format."""
+
+import argparse
+
+import numpy as np
+import pandas as pd
+
+parser = argparse.ArgumentParser(description=__doc__)
+parser.add_argument('-f', '--sj_file', help="STAR SJ.out.tab output file")
+args = parser.parse_args()
+
+sj_file = pd.read_csv(args.sj_file, delimiter='\t', header=None)
+
+# BED12 consists of 12 columns:
+header = ['chromosome',
+          'sj_start',
+          'sj_end',
+          'sj_name',
+          'score',
+          'strand',
+          'thick_start',
+          'thick_end',
+          'item_rgb',
+          'block_counts',
+          'block_sizes',
+          'block_starts']
+
+bed_file = pd.DataFrame(index=sj_file.index, columns=header)
+# 1: chromosome = first column from STAR SJ.out.tab
+bed_file.loc[:, 'chromosome'] = sj_file.iloc[:, 0].values
+
+# 2: SJ start (0-based) =
+# (first base of the intron (1-based) - maximum spliced alignment overhang) -1 (to recalculate to
+# 0 based system)
+bed_file.loc[:, 'sj_start'] = (sj_file.iloc[:, 1]) - (sj_file.iloc[:, 8]) - 1
+
+# 3: SJ end (0-based)
+# = (last base of the intron (1-based) + maximum spliced alignment overhang)
+bed_file.loc[:, 'sj_end'] = (sj_file.iloc[:, 2]) + (sj_file.iloc[:, 8])
+
+# 4: SJ name
+bed_file.loc[:, 'sj_name'] = 'JUNC000' + (sj_file.index + 1).astype(str)
+
+# 5: score = number of uniquely mapping reads crossing the junction
+bed_file.loc[:, 'score'] = sj_file.iloc[:, 6].values
+
+# 6: strand =  0: '.' (undefined) , 1: '+', 2: '-
+conditions = [
+    sj_file.iloc[:, 4] == 0,
+    sj_file.iloc[:, 4] == 1,
+    sj_file.iloc[:, 4] == 2
+]
+choices_strand = ['.', '+', '-']
+bed_file.loc[:, 'strand'] = np.select(conditions, choices_strand)
+
+# 7: thick start is the same as SJ start
+bed_file.loc[:, 'thick_start'] = (sj_file.iloc[:, 1]) - (sj_file.iloc[:, 8]) - 1
+
+# 8: thick end is the same as SJ end
+bed_file.loc[:, 'thick_end'] = (sj_file.iloc[:, 2]) + (sj_file.iloc[:, 8])
+
+# 9: item RGB = 255,0,0 (red color) for '-' strand, 0,0,255 (blue color) for '+' strand
+# and 0,0,0 (black) for undefined
+choices_rgb = ['0,0,0', '0,0,255', '255,0,0']
+bed_file.loc[:, 'item_rgb'] = np.select(conditions, choices_rgb)
+
+# 10: block counts = 2
+bed_file.loc[:, 'block_counts'] = '2'
+
+# 11: block sizes = maximum spliced alignment overhang, maximum spliced alignment overhang
+bed_file.loc[:, 'block_sizes'] = ((sj_file.iloc[:, 8]).astype(str) +
```

Comment:
Again, better than doing this:
    
    foobar = call(['a',
                   'b',
                   'c'])

rather use:

    foobar = call([
        'a',
        'b',
        'c',
    ])